core/services/job/models.go

```diff
@@ -850,47 +850,38 @@ type LiquidityBalancerSpec struct {
-	WorkflowName  string    `toml:"workflowName"`  // 10 byte plain text name
+	ID       int32  `toml:"-"`
+	Workflow string `toml:"workflow"` // the yaml representation of the workflow
+	// fields derived from the yaml spec, used for indexing the database
+	// note: i tried to make these private, but translating them to the database seems to require them to be public
+	WorkflowID    string    `toml:"-" db:"workflow_id"`    // Derived. Do not modify. the CID of the workflow.
+	WorkflowOwner string    `toml:"-" db:"workflow_owner"` // Derived. Do not modify. the owner of the workflow.
+	WorkflowName  string    `toml:"-" db:"workflow_name"`  // Derived. Do not modify. the name of the workflow.
 	CreatedAt     time.Time `toml:"-"`
 	UpdatedAt     time.Time `toml:"-"`
 }
 
 var (
-	ErrInvalidWorkflowID    = errors.New("invalid workflow id")
-	ErrInvalidWorkflowOwner = errors.New("invalid workflow owner")
-	ErrInvalidWorkflowName  = errors.New("invalid workflow name")
+	ErrInvalidWorkflowID       = errors.New("invalid workflow id")
+	ErrInvalidWorkflowYAMLSpec = errors.New("invalid workflow yaml spec")
 )
 
 const (
-	workflowIDLen = 64 // conveniently the same length as a sha256 hash
-	// owner and name are constrained the onchain representation in [github.com/smartcontractkit/chainlink-common/blob/main/pkg/capabilities/consensus/ocr3/types/Metadata]
-	workflowOwnerLen = 40 // hex string representation of 20 bytes
-	workflowNameLen  = 10 // plain text name
+	workflowIDLen = 64 // sha256 hash
 )
 
-// Validate checks the length of the workflow id, owner and name
-// that latter two are constrained by the onchain representation in [github.com/smartcontractkit/chainlink-common/blob/main/pkg/capabilities/consensus/ocr3/types/Metadata]
+// Validate checks the workflow spec for correctness
 func (w *WorkflowSpec) Validate() error {
-	if len(w.WorkflowID) != workflowIDLen {
-		return fmt.Errorf("%w: incorrect length for id %s: expected %d, got %d", ErrInvalidWorkflowID, w.WorkflowID, workflowIDLen, len(w.WorkflowID))
-	}
-
-	w.WorkflowOwner = strings.TrimPrefix(w.WorkflowOwner, "0x")
-	_, err := hex.DecodeString(w.WorkflowOwner)
+	s, err := pkgworkflows.ParseWorkflowSpecYaml(w.Workflow)
 	if err != nil {
-		return fmt.Errorf("%w: expected hex encoding got %s: %w", ErrInvalidWorkflowOwner, w.WorkflowOwner, err)
-	}
-	if len(w.WorkflowOwner) != workflowOwnerLen {
-		return fmt.Errorf("%w: incorrect length for owner %s: expected %d, got %d", ErrInvalidWorkflowOwner, w.WorkflowOwner, workflowOwnerLen, len(w.WorkflowOwner))
+		return fmt.Errorf("%w: failed to parse workflow spec %s: %w", ErrInvalidWorkflowYAMLSpec, w.Workflow, err)
 	}
+	w.WorkflowOwner = strings.TrimPrefix(s.Owner, "0x") // the json schema validation ensures it is a hex string with 0x prefix, but the database does not store the prefix
+	w.WorkflowName = s.Name
+	w.WorkflowID = s.CID
 
-	if len(w.WorkflowName) != workflowNameLen {
```

Comment:
Shouldn't we validate that the name is <= 10 chars? And only has alphanum chars? Does that happen in the schema validator now?

Apart from that we could either pad it to 10 chars right away or, alternatively, wait for the very last moment. In the latter case we could pad it here: https://github.com/smartcontractkit/chainlink/blob/develop/core/services/relay/evm/cap_encoder.go#L122

Reply:
same about validation.

padding in happening here is that the wrong place? https://github.com/smartcontractkit/chainlink-common/blob/c1110605a9906fcc64f0c105810c1835d6702075/pkg/capabilities/consensus/ocr3/types/aggregator.go#L43

that was the only use of the name that i could find and the closest place to do the padding to the actual serialization

Reply:
Padding over there is fine, thanks!